src/miner.h

```diff
@@ -160,7 +160,7 @@ class BlockAssembler
 public:
     BlockAssembler(const CChainParams& chainparams);
     /** Construct a new block template with coinbase to scriptPubKeyIn */
-    CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn);
+    std::unique_ptr<CBlockTemplate> CreateNewBlock(const CScript& scriptPubKeyIn);
```

Comment:
Converting from unique_ptr to shared_ptr is relatively cheap (specifically, does not require copying the object).